### PR TITLE
Rewrite `StandardUpdater` tests with pytest style assertion

### PR DESCRIPTION
**Pull request by niboshi**
(+ rename tests + added some comments)

### CODE REVIEW

File: tests/chainer_tests/training_tests/updaters_tests/test_standard_updater.py [hunk 1]
```diff
@@ -70,55 +70,53 @@ def setUp(self):
...
+        assert self.updater.loss_func is None
+        assert self.updater.iteration == 0
 
     def test_epoch(self):
-        self.assertEqual(self.updater.epoch, 1)
+        assert self.updater.epoch == 1
 
     def test_new_epoch(self):
-        self.assertTrue(self.updater.is_new_epoch)
+        assert self.updater.is_new_epoch is True
```
Comment (asi1024) [inline tests/chainer_tests/training_tests/updaters_tests/test_standard_updater.py]:
```suggestion
        assert self.updater.is_new_epoch
```

File: tests/chainer_tests/training_tests/updaters_tests/test_standard_updater.py [hunk 2]
```diff
@@ -70,55 +70,53 @@ def setUp(self):
...
-        self.assertEqual(self.updater.iteration, 1)
-        self.assertEqual(self.optimizer.epoch, 1)
-        self.assertEqual(self.iterator.next_called, 1)
+        assert self.updater.iteration == 1
+        assert self.optimizer.epoch == 1
+        assert self.iterator.next_called == 1
 
     def test_use_auto_new_epoch(self):
-        self.assertTrue(self.optimizer.use_auto_new_epoch)
+        assert self.optimizer.use_auto_new_epoch is True
```
Comment (asi1024) [inline tests/chainer_tests/training_tests/updaters_tests/test_standard_updater.py]:
```suggestion
        assert self.optimizer.use_auto_new_epoch
```

File: tests/chainer_tests/training_tests/updaters_tests/test_standard_updater.py (hunk 1 above)
Comment (niboshi) [reply tests/chainer_tests/training_tests/updaters_tests/test_standard_updater.py]: I think `is True` is better for testing.
`assert self.updater.is_new_epoch` would succeed if it is `5` or a function or anything else.

Comment (asi1024) [reply tests/chainer_tests/training_tests/updaters_tests/test_standard_updater.py]: OK, I understand.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**